### DISCUSSION TITLE
Negative Jacobian - contact in cylindrical tank

### DISCUSSION THREAD

**Discussion by OsamaAbdelaleim** [idaholab/moose#16616] [Q&A Modules: Solid mechanics] [answered]
I am building a model of a cylindrical tank filled with water in 3D. Frictionless contact is defined at all surfaces between the water and the tank. When running the input file attached (I replaced the water constitutive model I have with a simple elastic model), it fails in the first time step stating that there is a negative jacobian in a Quad 4 element. I have double checked the mesh file several times and it only contains hex 8 elements. I am wondering how to fix this issue for this specific geometry. 

I have tried different geometry and it only works when the thickness of the tank is increased. Also, it works if the contact block is commented out.
[3D_tank_error.zip](https://github.com/idaholab/moose/files/5765783/3D_tank_error.zip)

**Comment by jiangwen84** [comment, marked as answer]
Based on the error message, it looks like the error is related to the bottom contact surface (coord_z = -0.496 should be your bottom surface). Secondly, the error says that ''ERROR: negative Jacobian -2.06795e-25 at point (x,y,z)=(-0.0149897, -0.10797,   -0.496) in element 4294967295". That ID number is wired to me. 

I took a closer look at your mesh, and I believe there are some issues with your tank block mesh. To me, the bottom and the side are separate meshes, and not connected correctly. 

<img width="1157" alt="Screen Shot 2021-01-04 at 5 03 54 PM" src="https://user-images.githubusercontent.com/11078681/103592037-1f853900-4eaf-11eb-8d95-64316c1ff15f.png">
